pkg/transfer/local/pull.go

```diff
@@ -152,20 +160,28 @@ func (ts *localTransferService) pull(ctx context.Context, ir transfer.ImageFetch
 
 	// TODO: Should available platforms be a configuration of the service?
 	// First find suitable platforms to unpack into
-	//if unpacker, ok := is.
+	//If image storer is also an unpacker type, i.e implemented UnpackPlatforms() func
 	if iu, ok := is.(transfer.ImageUnpacker); ok {
 		unpacks := iu.UnpackPlatforms()
 		if len(unpacks) > 0 {
 			uopts := []unpack.UnpackerOpt{}
+			//Only unpack if requested unpackconfig matches default/supported unpackconfig
 			for _, u := range unpacks {
-				uopts = append(uopts, unpack.WithUnpackPlatform(u))
+				for _, du := range ts.config.UnpackConfigs {
+					if u.Platform.Match(du.Platform) && u.SnapshotterKey == du.Snapshotter {
+						uopts = append(uopts, unpack.WithUnpackPlatform(u))
```

Comment:
Use `du` here instead of `u`, for the reason I mentioned above, but we might also have other configuration in the future that the client wouldn't be expected to pass up. Examples may be running a shim instance to perform the unpack, that configuration could be daemon side and not something the client would be expected to have configured.

Reply:
If the condition is true which means `du` and `u` are the same.  Aren't they interchangeable here?

Reply:
We will want du to store additional default information that we wouldn't expect to be passed from the client.

Reply:
That makes sense, update in the coming commit.